### PR TITLE
Separate WaitGroup

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

File: core/utils/wait_group.hpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// TODO(MBkkt) Considered to replace with YACLib
+struct WaitGroup {
+  explicit WaitGroup(size_t counter = 0) noexcept : counter_{counter + 1} {}
+
+  void Add(size_t counter = 1) noexcept {
+    counter_.fetch_add(counter, std::memory_order_relaxed);
+  }
+
+  void Done(size_t counter = 1) noexcept {
+    if (counter_.fetch_sub(counter, std::memory_order_acq_rel) == 1) {
```
Comment (Dronplane) [inline core/utils/wait_group.hpp:40]:
```suggestion
    if (counter_.fetch_sub(counter, std::memory_order_acq_rel) == counter) {
```
?

**Review by Dronplane — CHANGES_REQUESTED** [review]
Why not use "counter*2" as delta values for `Add` and `Done`? And keep "1" as wait indicator. That will give us always accurate `Count`.

**Comment by MBkkt** [comment]
> Why not use "counter*2" as delta values for Add and Done? And keep "1" as wait indicator. That will give us always accurate Count.

We don't need "accurate" count.

File: core/utils/wait_group.hpp (hunk 1 above)
Comment (MBkkt) [reply core/utils/wait_group.hpp:40]: It's incorrect

**Comment by Dronplane** [comment]
> > Why not use "counter*2" as delta values for Add and Done? And keep "1" as wait indicator. That will give us always accurate Count.
> 
> We don't need "accurate" count.

Why we need method that intentionally lies?

**Comment by MBkkt** [comment]
> Why we need method that intentionally lies?

it's not lies

**Comment by Dronplane** [comment]
https://172.26.1.0/view/PR/job/iresearch-debug-pr/211/
https://172.26.1.0/view/PR/job/iresearch-pr/997/

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**